listenbrainz/webserver/static/js/src/types.d.ts

```diff
@@ -32,20 +32,25 @@ interface AdditionalInfo {
   work_mbids?: Array<string> | null;
 }
 
-declare type Listen = {
+declare type BaseListenFormat = {
   listened_at: number;
-  listened_at_iso?: string | null;
-  playing_now?: boolean | null;
   user_name?: string | null;
   track_metadata: {
     artist_name: string;
     release_name?: string | null;
     track_name: string;
     additional_info?: AdditionalInfo;
   };
+};
+
+declare type Listen = BaseListenFormat & {
+  listened_at_iso?: string | null;
+  playing_now?: boolean | null;
   score?: number;
 };
 
+declare type Recommendation = BaseListenFormat;
```

Comment:
That'll work nicely, and means you can easily extend the type with more fields as you need in the future, without having to change anything else( like we're gonna have to do for BP).
Yay future-proofing !